### PR TITLE
Add v1beta3 cut of builds

### PR DESCRIPTION
**Pull request by smarterclayton**
```
$ osc get bc/ruby-sample-build -o yaml --api-version=v1beta3
apiVersion: v1beta3
kind: BuildConfig
metadata:
  creationTimestamp: 2015-04-29T18:35:45Z
  labels:
    name: ruby-sample-build
  name: ruby-sample-build
  namespace: default
  resourceVersion: "128"
  selfLink: /osapi/v1beta3/namespaces/default/buildconfigs/ruby-sample-build
  uid: 8c196c95-ee9e-11e4-a80f-7831c1b76042
spec:
  output:
    to:
      kind: ImageStream
      name: origin-ruby-sample
  resources: {}
  source:
    git:
      uri: git://github.com/openshift/ruby-hello-world.git
    type: Git
  strategy:
    dockerStrategy:
      from:
        kind: ImageStreamTag
        name: ruby-20-centos7:latest
    type: Docker
  triggers:
  - github:
      secret: secret101
    type: github
  - generic:
      secret: secret101
    type: generic
  - imageChange:
      lastTriggeredImageID: openshift/ruby-20-centos7:latest
    type: imageChange
status:
  lastVersion: 1
```

```
apiVersion: v1beta3
kind: Build
metadata:
  creationTimestamp: 2015-04-29T18:35:47Z
  labels:
    buildconfig: ruby-sample-build
    name: ruby-sample-build
  name: ruby-sample-build-1
  namespace: default
  resourceVersion: "142"
  selfLink: /osapi/v1beta3/namespaces/default/builds/ruby-sample-build-1
  uid: 8d523608-ee9e-11e4-a80f-7831c1b76042
spec:
  output:
    to:
      kind: ImageStream
      name: origin-ruby-sample
  resources: {}
  source:
    git:
      uri: git://github.com/openshift/ruby-hello-world.git
    type: Git
  strategy:
    dockerStrategy:
      from:
        kind: DockerImage
        name: openshift/ruby-20-centos7:latest
    type: Docker
status:
  config:
    kind: BuildConfig
    name: ruby-sample-build
    namespace: default
  duration: 695000000000
  phase: Running
  startTimestamp: 2015-04-29T18:35:53Z
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
@bparees @csrwng @derekwaynecarr review please

File: pkg/build/api/v1beta3/conversion.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+		func(in *newer.BuildOutput, out *BuildOutput, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		func(in *BuildOutput, out *newer.BuildOutput, s conversion.Scope) error {
+			return s.DefaultConvert(in, out, conversion.IgnoreMissingFields)
+		},
+		// Rename ImageRepositoryRef to From
+		func(in *newer.ImageChangeTrigger, out *ImageChangeTrigger, s conversion.Scope) error {
+			// Note we lose the From/Image data in the ICT here if it was present,
+			// that data no longer has any value.
```
Comment (bparees) [inline pkg/build/api/v1beta3/conversion.go]: this isn't where we lose that data.  we lost it during v1beta1->internal conversion, not here where both internal and v1beta3 don't have that info.

File: pkg/build/api/v1beta3/types.go [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+// the status of the execution and a reference to the Pod which executed the build.
+type Build struct {
+	kapi.TypeMeta   `json:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is all the inputs used to execute the build.
+	Spec BuildSpec `json:"spec,omitempty"`
+
+	// Status is the current status of the build.
+	Status BuildStatus `json:"status,omitempty"`
```
Comment (bparees) [inline pkg/build/api/v1beta3/types.go]: if we're doing all this restructuring, don't we ideally want to go ahead and make the internal representation match this?

Comment (smarterclayton) [inline pkg/build/api/v1beta3/types.go]: After 3.0.  The internal refactor is extremely expensive.  Let's focus on features and completeness.

----- Original Message -----

Comment (bparees) [inline pkg/build/api/v1beta3/types.go]: +1, just figured i better ask.

**Comment by bparees** [comment]
i think that's it for me, looks good.

**Comment by csrwng** [comment]
LGTM

**Comment by smarterclayton** [comment]
We can continue to iterate on names - v1beta3 will not be marked official yet.

**Comment by smarterclayton** [comment]
[merge]

**Comment by derekwaynecarr** [comment]
LGTM even though this merged already.  It addresses my previous comments so I will close the issue I had opened on this

**Comment by derekwaynecarr** [comment]
Fixes https://github.com/openshift/origin/issues/1848

Is there a plan to support a subresource for /status updates ?

**Comment by smarterclayton** [comment]
If we need it.  Right now status is managed by the API

**Pull request merged**